### ISSUE TITLE
Native `quarkusBuild` fails with 3.0.0.CR2

### ISSUE THREAD

**Issue by shakhar** [quarkusio/quarkus#32509]
### Describe the bug

When running `./gradlew :apps:bianka-daemon:quarkusBuild -Dquarkus.package.type=native` with Quarkus 3.0.0.CR2, 
we get the following error:
```
========================================================================================================================
GraalVM Native Image: Generating 'bianka-daemon-0.1.0-runner' (executable)...
========================================================================================================================

[1/7] Initializing...                                                                                    (0.0s @ 0.63GB)
Fatal error: java.lang.NullPointerException: Cannot invoke "String.split(String)" because the return value of "java.lang.System.getenv(String)" is null
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker.lookupSearchPath(CCompilerInvoker.java:512)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker.getCCompilerPath(CCompilerInvoker.java:525)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker.getCCompilerInfo(CCompilerInvoker.java:371)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker.<init>(CCompilerInvoker.java:70)
------------------------------------------------------------------------------------------------------------------------
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker$DarwinCCompilerInvoker.<init>(CCompilerInvoker.java:275)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.codegen.CCompilerInvoker.create(CCompilerInvoker.java:82)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:910)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:575)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:535)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:403)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:580)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
                        0.6s (8.6% of total time) in 14 GCs | Peak RSS: 1.47GB | CPU load: 2.52
========================================================================================================================
Failed generating 'bianka-daemon-0.1.0-runner' after 7.0s.
Error: Image build request failed with exit status 1
> Task :apps:bianka-daemon:quarkusAppPartsBuild FAILED
```
We found that the error comes from [here](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/c/codegen/CCompilerInvoker.java#L519) because `System.getenv("PATH")` is null.
We suspect the issue related to [this](https://github.com/quarkusio/quarkus/commit/f10a208b09a8b50f48c6724586876e74436ecacd#diff-45931aed8a77981a5dfea9271e0c328b00b0d84d27dc67814dadaaa1570cb496R120) commit and more specifically to [this](https://github.com/quarkusio/quarkus/commit/f10a208b09a8b50f48c6724586876e74436ecacd#diff-c7c4670b6f52fa54bf95459cc59308526b7b4331fe8bfb512245ba93413344f5R160-R178) code block. We suspect that the new isolated process does not get all System's environment variables.

Note that when we ran the native-image command directly (not in the context of `quarkusAppPartsBuild` gradle task), the build worked.

### Output of `uname -a` or `ver`

Darwin ShakharArad 22.3.0 Darwin Kernel Version 22.3.0: Mon Jan 30 20:39:46 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6020 arm64

### Output of `java -version`

openjdk version "17.0.6" 2023-01-17 OpenJDK Runtime Environment GraalVM CE 22.3.1 (build 17.0.6+10-jvmci-22.3-b13) OpenJDK 64-Bit Server VM GraalVM CE 22.3.1 (build 17.0.6+10-jvmci-22.3-b13, mixed mode, sharing)

### GraalVM version (if different from Java)

22.3.1

### Quarkus version or git rev

3.0.0.CR2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

------------------------------------------------------------ Gradle 7.3.3 ------------------------------------------------------------  Build time:   2021-12-22 12:37:54 UTC Revision:     6f556c80f945dc54b50e0be633da6c62dbe8dc71  Kotlin:       1.5.31 Groovy:       3.0.9 Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021 JVM:          17.0.6 (GraalVM Community 17.0.6+10-jvmci-22.3-b13) OS:           Mac OS X 13.2.1 aarch64

**Comment by geoand** [comment]
Thanks for reporting.

Can you please check and see if the problem persists using the latest snapshot? 
I am asking because there have been a few relevant issues that have already been fixed.

**Comment by shakhar** [comment]
Tried to upgrade to 3.0.0.Final and now my gradle fails to sync with the following error:
```
* What went wrong:
Could not create task ':libs:bianka-common:test'.
> 'void org.gradle.api.tasks.testing.Test.notCompatibleWithConfigurationCache(java.lang.String)
```

**Comment by shakhar** [comment]
I think it something related to [this](https://github.com/quarkusio/quarkus/pull/32508) PR, I'll try to ensure we don't violate [Gradle build cache requirements](https://docs.gradle.org/8.0.2/userguide/configuration_cache.html#config_cache:requirements)

**Comment by Eng-Fouad** [comment]
I got the same error. Is there a workaround?

**Comment by geoand** [comment]
You can try using 3.0.0.Final (you need to use `io.quarkus` as the BOM )

**Comment by Eng-Fouad** [comment]
> You can try using 3.0.0.Final (you need to use `io.quarkus` as the BOM )

Now I am getting this error:

```
Error: Classes that should be initialized at run time got initialized during image building:
 jdk.internal.net.http.common.Utils the class was requested to be initialized at run time (for reading properties at run time). To see why jdk.internal.net.http.common.Utils got initialized use --trace-class-initialization=jdk.internal.net.http.common.Utils
To see how the classes got initialized, use --trace-class-initialization=jdk.internal.net.http.common.Utils
com.oracle.svm.core.util.UserError$UserException: Classes that should be initialized at run time got initialized during image building:
 jdk.internal.net.http.common.Utils the class was requested to be initialized at run time (for reading properties at run time). To see why jdk.internal.net.http.common.Utils got initialized use --trace-class-initialization=jdk.internal.net.http.common.Utils
To see how the classes got initialized, use --trace-class-initialization=jdk.internal.net.http.common.Utils
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:73)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ProvenSafeClassInitializationSupport.checkDelayedInitialization(ProvenSafeClassInitializationSupport.java:273)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.duringAnalysis(ClassInitializationFeature.java:164)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$10(NativeImageGenerator.java:748)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:85)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$11(NativeImageGenerator.java:748)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine.runAnalysis(AbstractAnalysisEngine.java:162)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:745)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:578)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:535)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:403)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:580)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
```

**Comment by geoand** [comment]
Any chance you can share a project that exhibits this behavior?

**Comment by Eng-Fouad** [comment]
My bad, I was trying few things and I had this line inside jax-rs resource:

    private static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();

After removing it, everything works fine.

**Comment by geoand** [comment]
Lovely, thanks

**Comment by geoand** [comment]
@snazy I am going to close this one as fixed by your changes in `3.0.0.Final`
